tests/resolve_message_reference.rs

```diff
@@ -73,3 +72,18 @@ bar = { foo } Bar
     let value = ctx.get_message("foo").and_then(|msg| ctx.format(msg, None));
     assert_eq!(value, Some("Foo ___".to_string()));
 }
+
+#[test]
+fn message_reference_multiple() {
+    let mut ctx = MessageContext::new(&["x-testing"]);
+
+    ctx.add_messages(
+        "
+foo = Foo
+bar = { foo } Bar { foo }
+",
+    );
+
+    let value = ctx.get_message("bar").and_then(|msg| ctx.format(msg, None));
+    assert_eq!(value, Some("Foo Bar Foo".to_string()));
+}
```

Comment:
Can you add a test that shows that we're breaking out of a cyclic reference?

```
foo = { bar }
bar = { foo }
```

Reply:
Sure thing